documentation/sphinx/source/bulkload-user.rst

```diff
@@ -0,0 +1,155 @@
+
+    fdb> bulkload history
+    Job b20ce68b03a28d654ee948ca4b5d859a submitted at 1741926085.210577 for range { begin=  end=\xff }. The job ran for 0.162005 mins and exited with status Complete.
+
+You can also clear 'all' history or by selectively remove jobs from the history list by 'id'.
+
+.. _blobstore:
+
+BulkDump/BulkLoad and S3
+========================
+
+In the above we illustrate dumping to a directory on the local filesystem.
+It is also possible to :command:`BulkDump` to, and
+:command:`BulkLoad` from, amazon's `S3 <https://aws.amazon.com/s3/>`_.
+Rather than reference a directory when dumping or loading,
+instead we make use of the fdb 'blobstore' url, or 'backup' url as it is also known, described in
+`Backup, Restore, and Replication for Disaster Recovery <https://apple.github.io/foundationdb/backups.html#backup-urls>`_.
+All backup configurations pertaining to 'S3' -- such as 'BLOBSTORE_CONCURRENT_UPLOADS', 'HTTP_VERBOSE_LEVEL',
+and 'BLOBSTORE_ENCRYPTION_TYPE' including how we specify credentials to 'S3'-- apply when running bulkload
+against 'S3' since bulkload uses the same underlying 'S3' accessing machinery.
+
+For illustration, let the 'S3' bucket that we want to dump be called
+'backup-123456789-us-west-2' and that this bucket is the 'us-west-2' amazon region.
+Let the 'prefix' that we want our dump to have in 'S3' be 'bulkload/test'. Then, in
+accordance with `Backup URLs <https://apple.github.io/foundationdb/backups.html#backup-urls>`_,
+our resulting url will be::
+
+    blobstore://@backup-123456789-us-west-2.s3.us-west-2.amazonaws.com/bulkload/test?bucket=backup-123456789-us-west-2&region=us-west-2
+ 
+Presuming the fdb cluster has been setup with the appropriate `blob credentials <https://apple.github.io/foundationdb/backups.html#blob-credential-files>`_ and 'mTLS' -- a site-specific affair -- below is how we'd dump a cluster to 'S3'::
+
+    fdb> bulkdump mode on
+    fdb> bulkdump dump "" \xff blobstore://@backup-123456789-us-west-2.s3.us-west-2.amazonaws.com/bulkload/test?bucket=backup-123456789-us-west-2&region=us-west-2
+    fdb> bulkdump status
+    ...
+
+Once :command:`status` reports 'No bulk dumping job is running', inspect the dumped dataset in 'S3' via your aws console or `aws s3 command-line tool <https://aws.amazon.com/cli/>`_. The job can take minutes or hours dependent on the amount of data hosted by your cluster.
+
+To load from 'S3' (presuming an empty cluster and presuming a previous dump whose 'id' is '123456789')::
+
+    fdb> bulkload mode on
+    fdb> bulkload load 123456789 "" \xff blobstore://@backup-123456789-us-west-2.s3.us-west-2.amazonaws.com/bulkload/test?bucket=backup-123456789-us-west-2&region=us-west-2
+    fdb> bulkload status
+    ...
+
+.. _troubleshooting:
+
+Troubleshooting
+===============
+
+:command:`BulkLoad` and :command:`BulkDump` require '--knob_shard_encode_location_metadata=1'.
+
+As for backup, enable '--knob_http_verbose_level 10' to debug connection issues.
+
```

Comment:
Search DDBulkLoad*, SSBulkLoad*, DDBulkDump*, SSBulkDump*, S3Client* in trace events to see more details.

Reply:
Good idea. Added.